pkg/controller/common/helpers_test.go

```diff
@@ -380,3 +381,35 @@ func TestRemoveIgnDuplicateFilesAndUnits(t *testing.T) {
 
 	assert.Equal(t, expectedIgn2Config, convertedIgn2Config)
 }
+
+func TestDiffMachineConfigFilesForPool(t *testing.T) {
+
+	var testIgn3ConfigOld ign3types.Config
+	var testIgn3ConfigNew ign3types.Config
+
+	oldTempFile := helpers.NewMachineConfigFile("/etc/kubernetes/kubelet-ca.crt", "oldcertificates", 420)
+	newTempFile := helpers.NewMachineConfigFile("/etc/kubernetes/kubelet-ca.crt", "newcertificates", 420)
+
+	// Make an "old" config with the existing file in it
+	testIgn3ConfigOld.Ignition.Version = "3.2.0"
+	testIgn3ConfigOld.Storage.Files = append(testIgn3ConfigOld.Storage.Files, *oldTempFile)
```

Comment:
We have a helper for creating MachineConfigs (https://github.com/openshift/machine-config-operator/blob/master/test/helpers/helpers.go#L37). Maybe I should add one with fewer parameters though :thinking: